ext/standard/var.c

```diff
@@ -868,13 +891,25 @@ static void php_var_serialize_class(smart_str *buf, zval *struc, zval *retval_pt
 		zend_string_free(prot_name);
 
 undef_prop:
-		php_var_serialize_string(buf, ZSTR_VAL(name), ZSTR_LEN(name));
-		php_var_serialize_intern(buf, &nval, var_hash);
 		php_error_docref(NULL, E_NOTICE,
 				"\"%s\" returned as member variable from __sleep() but does not exist", ZSTR_VAL(name));
+		if (val != NULL) {
+			ZEND_ASSERT(Z_TYPE_P(val) == IS_INDIRECT);
+			zend_property_info *info = zend_get_typed_property_info_for_slot(Z_OBJ_P(struc), Z_INDIRECT_P(val));
+			if (UNEXPECTED(info)) {
+				num_elements--;
+				continue;
+			}
+		}
```

Comment:
I don't think we should be special-casing typed properties here. Instead we should always omit the property if it doesn't exist, rather than setting it to a null value.